src/types/curve25519.rs

```diff
@@ -156,33 +153,22 @@ impl<'a> From<&'a ReusableSecret> for Curve25519PublicKey {
 }
 
 #[derive(Serialize, Deserialize)]
-pub(crate) struct Curve25519KeypairPickle {
-    secret: [u8; CURVE25519_SECRET_KEY_LEN],
-    public: [u8; Curve25519PublicKey::LENGTH],
-}
-
-impl Drop for Curve25519KeypairPickle {
-    fn drop(&mut self) {
-        self.secret.zeroize();
-    }
-}
+#[serde(transparent)]
+pub(crate) struct Curve25519KeypairPickle(Curve25519SecretKey);
```

Comment:
Sadly, I think we have to box this so that zeroization occurs correctly since `Curve25519SecretKey` is just a newtype-style wrapper for an array. If it's not boxed, then moving the pickle can make a copy of the array which won't get properly zeroized.

Reply:
There seem to be a couple of such problems, so this will be handled in a separate PR.